tests/backup_restore_test.php

```diff
@@ -0,0 +1,181 @@
+        return array(
+            'Instance Type ALL' => array(instance::TYPE_ALL),
+            'Instance Type Recording Only' => array(instance::TYPE_RECORDING_ONLY),
+            'Instance  Room Only' => array(instance::TYPE_ROOM_ONLY)
+        );
+    }
+
+    /**
+     * Instance type provider
+     *
+     * @return array
+     */
+    public function type_provider_with_recording(): array {
+        return array(
+            'Instance Type ALL' => array(instance::TYPE_ALL, 2),
+            'Instance Type Recording Only' => array(instance::TYPE_RECORDING_ONLY, 2),
+            'Instance  Room Only' => array(instance::TYPE_ROOM_ONLY, 2)
+        );
+    }
+
+    /**
+     * @dataProvider type_provider_with_recording
+     */
+    public function test_backup_restore_with_recordings($type, $nbrecordings) {
+        global $DB;
+        $this->resetAfterTest();
+        // This is for imported recording.
+        $othercourse = $this->generator->create_course();
+        $bbactivityothercourse = $this->generator->create_module(
+            'bigbluebuttonbn',
+            array('course' => $this->get_course()->id, 'type' => instance::TYPE_ALL),
+            ['visible' => true]);
+        $bbactivitysamecourse = $this->generator->create_module(
+            'bigbluebuttonbn',
+            array('course' => $this->get_course()->id, 'type' => instance::TYPE_ALL),
+            ['visible' => true]);
+
+        $bbactivity = $this->generator->create_module(
+            'bigbluebuttonbn',
+            array('course' => $this->get_course()->id, 'type' => $type, 'name' => 'BBB Activity'),
+            ['visible' => true]);
+
+        $bbbgenerator = $this->getDataGenerator()->get_plugin_generator('mod_bigbluebuttonbn');
+        $i = 0;
+        $recordingactivitylist = [
+            $bbactivityothercourse->id,
+            $bbactivitysamecourse->id
+        ];
+        if ($type === instance::TYPE_ALL) {
+            $recordingactivitylist[] = $bbactivity->id;
+        }
+        foreach ($recordingactivitylist as $bbbactivityid) {
+            foreach (range(1, $nbrecordings) as $rindex) {
+                $recordings[] = $bbbgenerator->create_recording(array_merge([
+                    'bigbluebuttonbnid' => $bbbactivityid,
+                    'name' => "PR Recording {$rindex}.{$i}"
+                ]));
+                $i++;
+            }
+        }
+        // Then import the recordings into the instance.
+        $instance = instance::get_from_instanceid($bbactivity->id);
+        foreach ($recordings as $rec) {
+            $rentity = recording::get_record(['id' => $rec->id]);
+            if ($rentity->get('bigbluebuttonbnid') != $instance->get_instance_id()) {
+                $rentity->create_imported_recording($instance);
+            }
+        }
+        $newcourseid = $this->backup_and_restore($this->get_course());
+        $newbbb = $DB->get_record('bigbluebuttonbn', ['course' => $newcourseid, 'name' => 'BBB Activity'], '*',
```

Comment:
Should ideally be split onto multiple lines. See coding style guidelines. We are moving towards PSR-12